jackson-core/src/main/java/io/micronaut/jackson/core/tree/MicronautTreeCodec.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2017-2021 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.jackson.core.tree;
+
+import com.fasterxml.jackson.core.*;
+import io.micronaut.json.JsonConfig;
+import io.micronaut.json.tree.JsonNode;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.util.*;
+
+public final class MicronautTreeCodec {
```

Comment:
Should this be annotated internal? Needs javadoc author/since tags

Reply:
This should have the same access as `JsonNode`. Right now, `JsonNode` is internal, but I forgot to make this internal as well. It might be better to make JsonNode public API though – it appears quite a bit in other APIs as well. Maybe it could be experimental?

Reply:
I would say mark everything public as `@Experimental` for the moment until we are satisfied with it